collections/ansible_collections/massopencloud/esi/roles/node/tasks/get_agent_names.yaml

```diff
@@ -0,0 +1,34 @@
+- name: Get agents
+  kubernetes.core.k8s_info:
+    kind: Agent
+    api_version: agent-install.openshift.io/v1beta1
+    namespace: "{{ manage_agents_namespace }}"
+  register: agents
+
+- name: Get matching agent names
+  ansible.builtin.set_fact:
+    agent_names: >-
+      {{
+        agent_names | default([]) + [
+          node_info.ports |
+          map(attribute='address') |
+          massopencloud.esi.match_agent_addresses(agents.resources)
+        ]
+      }}
+  loop: "{{ node_info_list }}"
+  loop_control:
+    loop_var: node_info
+    label: "{{ node_info.name }}"
+  when: node_info.ports | map(attribute='address') | massopencloud.esi.match_agent_addresses(agents.resources)
+
+- name: Get matching agent hostnames
+  ansible.builtin.set_fact:
+    agent_hostnames: >-
+      {{
+        agent_hostnames | default([]) + [node_info.name | lower]
+      }}
+  loop: "{{ node_info_list }}"
+  loop_control:
+    loop_var: node_info
+    label: "{{ node_info.name }}"
+  when: node_info.ports | map(attribute='address') | massopencloud.esi.match_agent_addresses(agents.resources)
```

Comment:
These two tasks loop the same way; is it possible to combine them and just return list of dicts?